pyomo/contrib/pyros/util.py

```diff
@@ -0,0 +1,683 @@
+        return obj
+    return fcn
+
+class grcsTerminationCondition(Enum):
+    '''
+    Enum class to describe termination conditions of the grcs algorithm
+    robust_optimal: the grcs algorithm returned with a robust_optimal solution under normal conditions
+    max_iter: the grcs algorithm could not identify a robust optimal solution within the specified max_iter.
+              Consider increasing the max_iter config param.
+    error: there was an error in the grcs solution procedure. Check the log file.
+    '''
+    robust_feasible = 0
+    robust_optimal = 1
+    robust_infeasible = 2
+    max_iter = 3
+    subsolver_error = 4
+    time_out = 5
+
+
+class SeparationStrategy(Enum):
+    all_violations = auto()
+    max_violation = auto()
+
+
+class SolveMethod(Enum):
+    local_solve = auto()
+    global_solve = auto()
+
+
+class ObjectiveType(Enum):
+    worst_case = auto()
+    nominal = auto()
+
+def model_is_valid(model):
+    '''
+    Possibilities:
+    Deterministic model has a single objective
+    Deterministic model has no objective
+    Deterministic model has multiple objectives
+    :param model: the deterministic model
+    :return: True if it satisfies certain properties, else False.
+    '''
+    objectives = list(model.component_data_objects(Objective))
+    for o in objectives:
+        o.deactivate()
+    if len(objectives) == 1:
+        '''
+        Ensure objective is a minimization. If not, change the sense.
+        '''
+        obj = objectives[0]
+
+        if obj.sense is not minimize:
+            sympy_obj = sympyify_expression(-obj.expr)
+            # Use sympy to distribute the negation so the method for determining first/second stage costs is valid
+            min_obj = Objective(expr=sympy2pyomo_expression(sympy_obj[1].simplify(), sympy_obj[0]))
+            model.del_component(obj)
+            model.add_component("obj",min_obj)
+        return True
+
+    elif len(objectives) > 1:
+        '''
+        User should deactivate all Objectives in the model except the one represented by the output of 
+        first_stage_objective + second_stage_objective
+        '''
+        return False
+    else:
+        '''
+        No Objective objects provided as part of the model, please provide an Objective to your model so that
+        PyROS can infer first- and second-stage objective.
+        '''
+        return False
+
+
+def turn_bounds_to_constraints(variable, model, config=None):
+    '''
+    Turn the variable in question's "bounds" into direct inequality constraints on the model.
+    :param variable: the variable with bounds to be turned to None and made into constraints.
+    :param model: the model in which the variable resides
+    :param config: solver config
+    :return: the list of inequality constraints that are the bounds
+    '''
+    if variable.lb is not None:
+        name = variable.name + "_lower_bound_con"
+        model.add_component(name, Constraint(expr=-variable <= -variable.lb))
```

Comment:
I have it this way because in the separation problem, we assume all constraints to separate are of the form g(x) <= 0 for simplicity in determining signs on violations and objective function senses.